pkg/util/node.go

```diff
@@ -75,3 +75,12 @@ func GetRootCgroupCurCFSPeriod(qosClass corev1.PodQOSClass) (int64, error) {
 	}
 	return strconv.ParseInt(strings.TrimSpace(rawContent), 10, 64)
 }
+
+func GetNodeAddress(node *corev1.Node, addrType corev1.NodeAddressType) string {
+	for _, address := range node.Status.Addresses {
+		if address.Type == addrType {
+			return address.Address
+		}
+	}
+	return ""
```

Comment:
Return an error here if there is no match. `return "", fmt.Errorf("no address matched types %v", addrType)`

Or you can validate the commad line input in the first place. Personally, I preffer the latter option.